### ISSUE TITLE
[Windows] [2.2.2] "Server is temporarily unavailable" forever (client doesn't seem to check server health)

### ISSUE THREAD

**Issue by schittli** [owncloud/client#5093]
### Expected behaviour
If the Client reports "Server is temporarily unavailable" then it should re-test the connection periodically. Especially because manually closing and restarting the Client resolves the problem.

### Actual behaviour
After some time, the Windows Client reports "Server is temporarily unavailable" - and it does not get out of this state.

### Steps to reproduce
1. Start Windows Client
2. Let it work / wait some time
3. the Windows Client reports "Server is temporarily unavailable"

### Server configuration
The Client should heal its connection for all server configurations

### Client configuration
Client version: 2.2.2 6192
Operating system: WIndows 10 x64
OS language: English

### Logs
It looks like OwnCloud Client has no setting for enabling any log :-(

**Comment by jedwards1211** [comment]
I'm seeing this in general on OSX -- if it says 'no connection to blah.com', and I make it log out and log back in, it still says 'no connection to blah.com'.  If the network goes down the client seems to be stuck thinking it's disconnected.  Even if some files fail to transfer -- there's no indication if or when the client is ever going to retry anything, and making it log out and log back in doesn't seem to change anything.  It's as if the client has cached the failure message and given up.

The only workaround is to completely remove the account from the client and add it back again, which is pretty inconvenient.

**Comment by guruz** [comment]
(Deleted and moved @jedwards1211 comment to  https://github.com/owncloud/client/issues/5096 )

**Comment by guruz** [comment]
I tried reproducing this by unplugging my USB network adapter and waiting until it shows "No connection to [...]". Then I replug it a bit later, after a while it reconnects. So all fine for me with 2.2.**3beta1**. 

I need to check how to get the "Server is temporarily unavailable" state. Was there a maintenance on the server? Or did the server otherwise return a lot of 503 errors? (Check the server access log and possibly owncloud.log)

We'd need a relevant log file from you to be able to dig into this. https://doc.owncloud.org/desktop/2.2/troubleshooting.html

**Comment by dragotin** [comment]
To get "Server temp. unavailable" you just need to set 
```
  'maintenance' => true,
```
in the servers `config.php`.

**Comment by schittli** [comment]
We've found an Configuration-Issue on the OwnCloud-Server and because of the missing Logging-Infomations on the Windows-OwnCloud-Client we can not say if it tried to reconnect :-(

We had the following Configuration Issue on the Server:
The LDAP query with the list of allowed users had a filter:
> (&(memberOf:1.2.840.113556.1.4.1941:=CN=xxx)(**sAMAccountName=%uid**))
This query must return a list of *all* allowed users, therefor it must be:
> (memberOf:1.2.840.113556.1.4.1941:=CN=xxx)
Actually, we use this:
> (&(|(objectclass=person)(objectclass=user))(memberOf:1.2.840.113556.1.4.1941:=CN=xxx))

If the LDAP query with the list of all allowed users filters for %uid, then:
- the LDAP query returns an empty list
- therefore, the OwnCloud Client can connect
- but the Server will notify that User xxx is not a valid user anymore - and it will block the user
- therefore, the OwnCloud Client stops the syncronization
- if the OwnCloud Client is restarted, then it works fine until the server blocks this user again

In short / summary:
- if the OwnCloud Client really automatically checks the Server Health, then everything is OK
- but the OwnCloud Client should log the "connection lost" and the "re-testing connection" events - at least if a user activates verbose logging :-)

Thanks a lot for all your great support!
kind regards, Tom

**Comment by guruz** [comment]
FYI The client does log this :) At least for me..

```
09-05 15:03:02:938 381396 OCC::AbstractNetworkJob::start: !!! OCC::PropfindJob created for "http://127.0.0.1" + "/" "OCC::ConnectionValidator"
09-05 15:03:02:979 381396 OCC::AbstractNetworkJob::slotFinished: void OCC::AbstractNetworkJob::slotFinished() QNetworkReply::NetworkError(ServiceUnavailableError) "Error transferring http://127.0.0.1:1240/remote.php/webdav/ - server replied: Service Unavailable" QVariant(int, 503)
09-05 15:03:02:979 381396 OCC::PropfindJob::finished: PROPFIND request *not* successful, http result code is 503 ""
09-05 15:03:02:980 381396 OCC::FolderMan::slotAccountStateChanged: Account "woboq@127.0.0.1:1240" disconnected or paused, terminating or descheduling sync folders
```

**Comment by guruz** [comment]
For the UI/UX improvements let's track this in https://github.com/owncloud/client/issues/5096
